packages/controllers/src/snaps/endowments/cronjob.test.ts

```diff
@@ -0,0 +1,168 @@
+      expect(specification).toStrictEqual({
+        permissionType: PermissionType.Endowment,
+        targetKey: SnapEndowments.Cronjob,
+        endowmentGetter: expect.any(Function),
+        allowedCaveats: [SnapCaveatType.SnapCronjob],
+      });
+
+      expect(specification.endowmentGetter()).toBeUndefined();
+    });
+  });
+
+  describe('cronjobCaveatMapper', () => {
+    it('returns a caveat value for the objects of cronjob specification', () => {
+      expect(
+        getCronjobCaveatMapper({
+          jobs: [
+            {
+              expression: '* * * * *',
+              request: {
+                method: 'exampleMethodOne',
+                params: ['p1'],
+              },
+            },
+            {
+              expression: '* * * * *',
+              request: {
+                method: 'exampleMethodTwo',
+                params: ['p1', 'p2', 'p3'],
+              },
+            },
+          ],
+        }),
+      ).toStrictEqual({
+        caveats: [
+          {
+            type: SnapCaveatType.SnapCronjob,
+            value: {
+              jobs: [
+                {
+                  expression: '* * * * *',
+                  request: {
+                    method: 'exampleMethodOne',
+                    params: ['p1'],
+                  },
+                },
+                {
+                  expression: '* * * * *',
+                  request: {
+                    method: 'exampleMethodTwo',
+                    params: ['p1', 'p2', 'p3'],
+                  },
+                },
+              ],
+            },
+          },
+        ],
+      });
+    });
+  });
+});
+
+describe('validateCronjobCaveat', () => {
+  it('should not throw an error when provided specification is valid', () => {
+    const caveat: Caveat<string, any> = {
+      type: SnapCaveatType.SnapCronjob,
+      value: {
+        jobs: [
+          {
+            expression: '* * * * *',
+            request: {
+              method: 'snapMethod',
+              params: [],
+            },
+          },
+        ],
+      },
+    };
+
+    expect(() => validateCronjobCaveat(caveat)).not.toThrow();
+  });
+
+  it('should throw if caveat has no proper value', () => {
+    const caveat: Caveat<string, any> = {
+      type: SnapCaveatType.SnapCronjob,
+      value: {},
+    };
+
+    expect(() => validateCronjobCaveat(caveat)).toThrow(
+      `Expected a plain object.`,
+    );
+  });
+
+  it('should throw an error when cron specification is missing', () => {
+    const caveat: Caveat<string, any> = {
+      type: SnapCaveatType.SnapCronjob,
+      value: {
+        jobs: [
+          {
+            expression: '* * * * *',
+            request: undefined as any,
+          },
+        ],
+      },
+    };
+
+    expect(() => validateCronjobCaveat(caveat)).toThrow(
+      'Expected a valid cronjob specification array.',
+    );
+  });
+});
+
+describe('CronjobCaveatSpecifications', () => {
+  describe('validator', () => {
```

Comment:
Can you test that this works with the object format too?

Reply:
Added it now.